### PR TITLE
Kokkos_ViewMapping.hpp: intel workaround for using statement

### PR DESCRIPTION
**Pull request by ndellingwood**
Temporarily replace using with typedef (and clang-tidy nolint marking)
  to workaround intel compiler issue
"error #3186: expected typedef declaration"

Address issue #3381

### CODE REVIEW

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 1]
```diff
@@ -2725,9 +2725,11 @@ struct ViewDataHandle<
                                    Kokkos::CudaUVMSpace>::value))
 #endif
                 && (!Traits::memory_traits::is_atomic))>::type> {
-  using value_type  = typename Traits::value_type;
-  using handle_type = typename Traits::value_type* KOKKOS_IMPL_ALIGN_PTR(
-      KOKKOS_MEMORY_ALIGNMENT);
+  using value_type = typename Traits::value_type;
+  // typedef work-around for intel compilers error #3186: expected typedef
+  // declaration
+  typedef typename Traits::value_type* KOKKOS_IMPL_ALIGN_PTR(
```
Comment (DavidPoliakoff) [inline core/src/impl/Kokkos_ViewMapping.hpp:2731]: Why `Traits::value_type*` when you alias value_type to Traits::value_type above? Could just be `value_type*`, no?

Comment (dalg24) [reply core/src/impl/Kokkos_ViewMapping.hpp:2731]: Sure

**Review by dalg24 — APPROVED** [review]

File: core/src/impl/Kokkos_ViewMapping.hpp (hunk 1 above)
Comment (ndellingwood) [reply core/src/impl/Kokkos_ViewMapping.hpp:2731]: Yeah, I didn't look carefully when I replaced the using with typedef, clang-format pulled in the `value_type` line 2728 in (no apparent change). I'll rebase the change in, thanks

**Review by masterleinad — APPROVED** [review]

**Review by DavidPoliakoff — APPROVED** [review]
My only comment can be addressed in a future PR, no need for it here

File: core/src/impl/Kokkos_ViewMapping.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/impl/Kokkos_ViewMapping.hpp:2731]: This does not need to be addressed at this time.  My preference goes to merging your PR as you opened it.

Comment (ndellingwood) [reply core/src/impl/Kokkos_ViewMapping.hpp]: Just rebased the cleanup in

**Comment by DavidPoliakoff** [comment]
Apparently failed on format check

**Comment by ndellingwood** [comment]
Huh, I ran clang-format locally (clang8). Let me try again...

**Comment by dalg24** [comment]
Retest this please

**Comment by ndellingwood** [comment]
I tried applying clang-format again but did not see a difference, I'm guessing my force-push messed up the CI? 

```bash
s994290:kokkos ndellin$ git log -1
commit 36662031444e10d36f4491f8852205cab15e109c (HEAD -> issue-3381, myfork/issue-3381)
Author: Nathan Ellingwood <ndellin@sandia.gov>
Date:   Mon Sep 14 12:57:33 2020 -0600

    Kokkos_ViewMapping.hpp: intel workaround for using statement
    
    Temporarily replace using with typedef (and clang-tidy nolint marking)
      to workaround intel compiler issue
    "error #3186: expected typedef declaration"
    
    Address issue #3381
s994290:kokkos ndellin$ clang-format -i core/src/Kokkos_Macros.hpp core/src/impl/Kokkos_ViewMapping.hpp
s994290:kokkos ndellin$ git status
On branch issue-3381
Your branch is up to date with 'myfork/issue-3381'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	cherry-pick-commits.txt

nothing added to commit but untracked files present (use "git add" to track)
s994290:kokkos ndellin$ clang --version
clang version 8.0.1 (tags/RELEASE_801/final)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /opt/local/libexec/llvm-8.0/bin
```

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 2]
```diff
@@ -2725,9 +2725,11 @@ struct ViewDataHandle<
...
 #endif
                 && (!Traits::memory_traits::is_atomic))>::type> {
-  using value_type  = typename Traits::value_type;
-  using handle_type = typename Traits::value_type* KOKKOS_IMPL_ALIGN_PTR(
-      KOKKOS_MEMORY_ALIGNMENT);
+  using value_type = typename Traits::value_type;
+  // typedef work-around for intel compilers error #3186: expected typedef
+  // declaration
+  typedef value_type* KOKKOS_IMPL_ALIGN_PTR(KOKKOS_MEMORY_ALIGNMENT)
+      handle_type;  // NOLINT(modernize-use-using)
```
Comment (dhollman) [inline core/src/impl/Kokkos_ViewMapping.hpp:2732]: Does this the pass clang-tidy check? We might have to use `NOLINTNEXTLINE` or something if not

**Review by dhollman — APPROVED** [review]

File: core/src/impl/Kokkos_ViewMapping.hpp (hunk 2 above)
Comment (ndellingwood) [reply core/src/impl/Kokkos_ViewMapping.hpp]: Just pushed the `NOLINTNEXTLINE` change

**Comment by ndellingwood** [comment]
@dalg24 @masterleinad is there something I need to do differently for the PR to address the CI failures? There are errors like:

OMPTARGET-clang
```
nvlink fatal   : Internal error: writing file
clang-10: error: nvlink command failed with exit code 1 (use -v to see invocation)
core/unit_test/CMakeFiles/KokkosCore_UnitTest_OpenMPTarget.dir/build.make:1528: recipe for target 'core/unit_test/KokkosCore_UnitTest_OpenMPTarget' failed
make[2]: *** [core/unit_test/KokkosCore_UnitTest_OpenMPTarget] Error 1
CMakeFiles/Makefile2:1475: recipe for target 'core/unit_test/CMakeFiles/KokkosCore_UnitTest_OpenMPTarget.dir/all' failed
make[1]: *** [core/unit_test/CMakeFiles/KokkosCore_UnitTest_OpenMPTarget.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

And elsewhere out-of-disk messages during linking attempts:
```
/usr/bin/ld: final link failed: No space left on device
collect2: error: ld returned 1 exit status
```

**Comment by dalg24** [comment]
Not your fault.  Both failing jobs ran on fetnat05 with `No space left on device`.  I was messing with the machine when these happen.  I will re-run.

Retest this please.
